MdePkg/Library/BaseRngLib/Rand/RdRand.c

```diff
@@ -70,6 +156,7 @@ ArchGetRandomNumber16 (
   OUT     UINT16  *Rand
   )
 {
+  ASSERT (mRdRandSupported);

```

Comment:
Library API GetRandomNumber16() doesn't describe this ASSERT condition. I suggest to return FALSE when mRdRandSupported is FALSE, don't add ASSERT here.

Reply:
Calling ArchGetRandomNumber despite ArchIsRngSupported returning FALSE is a very clear bug. ARM code atm will just crash (with an invalid instruction/mrs) if that happens.

Reply:
What @heatd says.  Note that this isn't `GetRandomNumber16` but `ArchGetRandomNumber16`.
 `GetRandomNumber16` calls `ArchGetRandomNumber16` but only after checking `mRdRandSupported`.

Reply:
+1 - the ASSERT() can only be triggered by an internal inconsistency, so it is not part of the library API

Reply:
I agree this change doesn't impact the public API behavior.